typed/union.go

```diff
@@ -105,38 +108,38 @@ type discriminator struct {
 	name string
 }
 
-func (d *discriminator) Set(m *value.Map, v discriminated) {
+func (d *discriminator) Set(m value.Map, v discriminated) {
 	if d == nil {
 		return
 	}
-	m.Set(d.name, value.StringValue(string(v)))
+	m.Set(d.name, value.NewValueInterface(string(v)))
 }
 
-func (d *discriminator) Get(m *value.Map) discriminated {
+func (d *discriminator) Get(m value.Map) discriminated {
 	if d == nil || m == nil {
 		return ""
 	}
-	f, ok := m.Get(d.name)
+	val, ok := m.Get(d.name)
 	if !ok {
 		return ""
 	}
-	if f.Value.StringValue == nil {
+	if !val.IsString() {
 		return ""
```

Comment:
I sure hope this is checked by the schema :/

Reply:
It's not that easy? I can't remember if we're checking for type aliases to strings. What do you suggest?

Reply:
I don't know, but if this is not enforced by the schema (which really should be able to follow aliases), then this is going to be a VERY mysterious failure mode for someone.

Reply:
Agreed, we should improve that